logstash-core/lib/logstash/pipeline.rb

```diff
@@ -38,6 +38,7 @@ def self.validate_config(config_str, settings = {})
     begin
       # There should be a better way to test this
       self.new(config_str, settings)
+      true
     rescue => e
       e.message
```

Comment:
well yes an no, at minimum I would introduce some clear documentation ... the return value interface here is just misleading and confusing as it return two kind of values. We need to provide clarity in order to avoid feature issues.

And actually there is only one user of this interface and is the one you just fix.

And yes, your fix fixed the issue for this change, but we could also help out on fixing the interface as is mostly part of why this happened.